### DISCUSSION TITLE
custom extension method polly

### DISCUSSION THREAD

**Discussion by FabioMarturanoJr** [App-vNext/Polly#1044] [General]
Hi,
I trying to create my on extension method to policy, is this possible ?
Because I'm failing miserably 

using the extension
![image](https://user-images.githubusercontent.com/78613817/221949269-d01cca8e-0b03-4f6b-adf9-679260a33188.png)

 creating 
![image](https://user-images.githubusercontent.com/78613817/221949405-74f3daa1-1d14-4d3e-816c-7513866c04b6.png)

**Comment by martincostello** [comment]
You can’t write extension methods for the type statically, only an _instance_ of that type.

This won’t work:

```csharp
Policy.MyExtensionMethod();
```

This will:

```csharp
var myPolicy = ...;
myPolicy.MyExtensionMethod();
```

You’ll need to either start from a custom static class:

```csharp
var policy = MyPolicy.HandleAndRetry(...);
```

Or extend the policy builder and extend from the _second_ line of what you’re trying to do in your extension method.

**Comment by FabioMarturanoJr** [comment]
it worked, creating the extension method from the PolicyBuilder,
thank you very much :)

**Comment by peter-csala** [comment]
As a side-note
```cs
if (policyResult.FinalException != null)
   throw policyResult.FinalException;

return policyResult;
```
IMHO it does not make too much sense. If the execution fails then it will throw an exception. If it succeeds then you have to unwrap the result from the `PolicyResult`.

I would suggest to simply prefer `ExecuteAsync` over `ExecuteAndCaptureAsync` in this particular case.
